### DISCUSSION TITLE
Improve discoverability of maps with additional content

### DISCUSSION THREAD

**Discussion by monochrome22** [ppy/osu#23511] [Ideas]
By additional content, I mean videos, storyboards, custom skins, custom hitsounds.

Currently, there is no way to know if a map has any additional content without entering gameplay. 
There should be search options like `video=1` `storyboard=1` `skin=1` `hitsound(s)=1`.
Additionally, icons could be added to song select, something like this:

![icons](https://github.com/ppy/osu/assets/122969285/f7b68a69-a724-4fd2-ba79-d735b6566623)
(Illustrative purpose only)

**Comment by aryoadhi** [comment]
I think it's great to implement for beatmaps with video and storyboards. Unsure for skin and hitsounds though, since there are beatmaps with very subtle skin changes and only uses one hitsound, namely `hitnormal` (see mania beatmaps)

**Reply by monochrome22** [reply]
Even if some beatmaps only have subtle changes, being able to search and distinguish them would still be better than nothing. Currently, if you are trying to find a map with certain custom skin elements, you will have to go through many maps, most of which aren't going to have any custom skin elements. If it was possible to search for maps with any skins elements, you would have a much higher chance of finding a map with the skin elements that you are looking for.
Maybe display a list of all the custom elements while hovering the skin or hitsound icon, though this might not look good.

**Reply by Walavouchey** [reply]
the same argument can be made for videos and storyboards. just because a map has video or storyboard content, doesn't mean that it's to any great extent. a large portion of "storyboarded" maps do *very* tiny things, such as adding one or two sprites or only having storyboard sound samples (keysounds)

i agree with being able to filter all of these as suggested

**Reply by peppy** [reply]
To add to the conversation, I'd like to see these kinds of flags as a potentially manual thing, only marking those with serious usage of the features. Maybe with automatic rules as a starting point then toggling maps which don't actually meet sane criteria from an end-product perspective.

**Reply by aryoadhi** [reply]
Is it possible like the genre/language field on beatmap listing, which can be edited by everyone but when ranked it's only editable by NATs/GMTs? It's kinda taxing to check every single maps, but I can see the potential for it.

**Comment by Theighlin** [comment]
Partly duplicate of #5752
